### PR TITLE
Initial implementation of authentication flow

### PR DESCRIPTION
**Pull request by MudassirQuraishi**
This commit introduces the initial setup and basic functionality for the authentication flow. Included are login, and signup, along with necessary backend configurations.

### CODE REVIEW

File: src/api/api.js
Comment (sourav0809) [inline src/api/api.js:1]: add some comments for apis

File: src/app/App.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+
+function App() {
+    const { isLoggedIn } = useSelector((state) => state.auth)
+    const [loading, setLoading] = useState(false);
+    return (
+        <>
+            {loading && <PageLoader />}
+            {!isLoggedIn && <AuthRoutes />}
```
Comment (sourav0809) [inline src/app/App.jsx]: check is logged in in single line use the ternary operator

File: src/components/auth/AuthHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import PropTypes from 'prop-types';
+const AuthHeader = ({ label, title }) => {
+    return (<>
+        <div className="w-full flex flex-col gap-y-4 items-center justify-center">
+            <h1 className="text-3xl font-semibold">{title}</h1>
+            <p className="text-muted-foreground  text-sm">{label}</p>
+        </div>
+    </>)
+}
+AuthHeader.propTypes = {
```
Comment (sourav0809) [inline src/components/auth/AuthHeader.jsx]: don't need this

File: src/components/auth/AuthLayout.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import PropTypes from 'prop-types'
+import { Menu } from "lucide-react"
+import { Link } from 'react-router-dom'
+import { useState } from 'react'
+import { CalendarCheck } from 'lucide-react'
+const AuthLayout = ({ children }) => {
+    const [state, setState] = useState(false)
+    const menus = [
```
Comment (sourav0809) [inline src/components/auth/AuthLayout.jsx]: make a separate file called header constant

File: src/components/auth/BackButton.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { Link } from 'react-router-dom'
+const BackButton = ({ label, path }) => {
+    return (<>
+        <Button variant="link" className="font-normal w-full" asChild size="sm">
+            <Link to={path}>{label}</Link>
+        </Button>
+
+    </>)
+}
+BackButton.propTypes = {
```
Comment (sourav0809) [inline src/components/auth/BackButton.jsx]: remove this

File: src/components/ui/button.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import * as React from "react"
+import { Slot } from "@radix-ui/react-slot"
+import { cva } from "class-variance-authority";
+
+import { cn } from "@/lib/utils"
+
+const buttonVariants = cva(
```
Comment (sourav0809) [inline src/components/ui/button.jsx:7]: should start with capital letter

File: src/components/auth/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            email: "",
+            password: ""
+        }
+    })
+    const onSubmit = async (data) => {
+        dispatch(logInAction(data, setIsSubmitting))
+    }
+    return (<>
+        <CardWrapper label="Welcome back, we missed you." title="Login" backButtonPath="/auth/signup" backButtonLabel="Don't have an account? Signup here">
+            <Form {...form}>
```
Comment (sourav0809) [inline src/components/auth/Login.jsx:36]: if possible try to make your own form

File: src/components/common/CardWrapper.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                    <BackButton label={backButtonLabel} path={backButtonPath} />
+                </CardFooter>
+            </Card>
+
+        </>
+    );
+};
+
+
+CardWrapper.propTypes = {
```
Comment (sourav0809) [inline src/components/common/CardWrapper.jsx]: remove this

File: src/routes/AuthRoutes.jsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+import Signup from "@/components/auth/Signup";
+import { Route, Routes } from "react-router-dom";
+
+const AuthRoutes = () => {
+    return (<>
+        <AuthLayout>
+            <Routes>
+                <Route path="/auth/signup" element={<Signup />} />
+                <Route path="/auth/login" element={<Login />} />
+            </Routes>
```
Comment (sourav0809) [inline src/routes/AuthRoutes.jsx:13]: no error routes present add "*" in the last

File: src/routes/UserRoutes.jsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { Routes, Route } from "react-router-dom"
+
+const UserRoutes = () => {
+    return (<>
+        <Routes>
+            <Route path="/home" />
+        </Routes>
```
Comment (sourav0809) [inline src/routes/UserRoutes.jsx:8]: no error route present

File: src/schema/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import { z } from 'zod';
+export const signupSchema = z.object({
```
Comment (sourav0809) [inline src/schema/index.js:2]: change the name

File: src/store/actions/authActions.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
+import { signupUser, loginUser } from '@/api/api';
+
+import {
+  setUserDetails,
+  setUserLoggedIn,
+  logOutUser,
+} from '../reducers/authSlice';
+
```
Comment (sourav0809) [inline src/store/actions/authActions.js]: add some comments

File: src/store/actions/authActions.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        email: data.email,
+      };
+      dispatch(setUserDetails(userDetails));
+      dispatch(setUserLoggedIn());
+
+      localStorage.setItem('token', data.token);
+    } catch (err) {
+      console.log(err);
+    }
+    setBtnLoader(false);
```
Comment (sourav0809) [inline src/store/actions/authActions.js]: make the button loader false in finally block

**Review by sourav0809 — CHANGES_REQUESTED** [review]

File: src/components/ui/button.jsx (hunk 5 above)
Comment (MudassirQuraishi) [reply src/components/ui/button.jsx:7]: These are shadcn's components,

File: src/components/auth/Login.jsx (hunk 6 above)
Comment (MudassirQuraishi) [reply src/components/auth/Login.jsx:36]: We have the option to use a UI Library

File: src/api/api.js
Comment (MudassirQuraishi) [reply src/api/api.js:1]: Lets do documentation at the end.

Comment (MudassirQuraishi) [reply src/api/api.js:1]: Added documentation for API

File: src/app/App.jsx (hunk 1 above)
Comment (MudassirQuraishi) [reply src/app/App.jsx]: Updated the logic here

File: src/components/auth/AuthHeader.jsx (hunk 2 above)
Comment (MudassirQuraishi) [reply src/components/auth/AuthHeader.jsx]: Removed Proptypes

File: src/components/auth/AuthLayout.jsx (hunk 3 above)
Comment (MudassirQuraishi) [reply src/components/auth/AuthLayout.jsx]: Moved the data to another file

File: src/components/auth/BackButton.jsx (hunk 4 above)
Comment (MudassirQuraishi) [reply src/components/auth/BackButton.jsx]: Removed proptypes

File: src/components/ui/button.jsx (hunk 5 above)
Comment (MudassirQuraishi) [reply src/components/ui/button.jsx:7]: Updated file names

File: src/components/common/CardWrapper.jsx (hunk 7 above)
Comment (MudassirQuraishi) [reply src/components/common/CardWrapper.jsx]: Removed Proptypes

File: src/store/actions/authActions.js (hunk 11 above)
Comment (MudassirQuraishi) [reply src/store/actions/authActions.js]: Better comments added

File: src/store/actions/authActions.js (hunk 12 above)
Comment (MudassirQuraishi) [reply src/store/actions/authActions.js]: update the try catch blocks

**Review by sourav0809 — APPROVED** [review]

**Pull request merged**